src/openrct2/peep/Peep.cpp

```diff
@@ -756,57 +756,51 @@ void peep_window_state_update(rct_peep * peep)
     }
 }
 
-void peep_pickup(rct_peep * peep)
+void rct_peep::Pickup()
 {
-    peep->RemoveFromRide();
-    peep->Invalidate();
+    RemoveFromRide();
+    Invalidate();
 
-    sprite_move(LOCATION_NULL, peep->y, peep->z, (rct_sprite *)peep);
-    peep->SetState(PEEP_STATE_PICKED);
-    peep->sub_state = 0;
+    sprite_move(LOCATION_NULL, y, z, (rct_sprite *)this);
+    SetState(PEEP_STATE_PICKED);
+    sub_state = 0;
 }
 
-void peep_pickup_abort(rct_peep * peep, sint32 old_x)
+void rct_peep::PickupAbort(sint32 old_x)
 {
-    if (!peep)
-        return;
-
-    if (peep->state != PEEP_STATE_PICKED)
+    if (state != PEEP_STATE_PICKED)
         return;
 
-    sprite_move(old_x, peep->y, peep->z + 8, (rct_sprite *)peep);
-    peep->Invalidate();
+    sprite_move(old_x, y, z + 8, (rct_sprite *)this);
+    Invalidate();
 
-    if (peep->x != (sint16)LOCATION_NULL)
+    if (x != (sint16)LOCATION_NULL)
     {
-        peep->SetState(PEEP_STATE_FALLING);
-        peep->action                     = 0xFF;
-        peep->special_sprite             = 0;
-        peep->action_sprite_image_offset = 0;
-        peep->action_sprite_type         = 0;
-        peep->path_check_optimisation                     = 0;
+        SetState(PEEP_STATE_FALLING);
+        action                     = 0xFF;
+        special_sprite             = 0;
+        action_sprite_image_offset = 0;
+        action_sprite_type         = 0;
+        path_check_optimisation    = 0;
     }
 
     gPickupPeepImage = UINT32_MAX;
 }
 
-bool peep_pickup_place(rct_peep * peep, sint32 x, sint32 y, sint32 z, bool apply)
+bool rct_peep::Place(sint32 posX, sint32 posY, sint32 posZ, bool apply)
```

Comment:
This should probably still keep `Pickup` in the function name.

Reply:
I had `PickupPlace` before, but found it more confusing that just `Place`.

Reply:
Well we should at least add a comment that this is only for use with picked up peeps. Don't want it confused with move.